### PR TITLE
[PFX-811] - Tune Intl action

### PR DESCRIPTION
**Pull request by mmason2-godaddy**
## Summary
- [Tune getIntlManager to import intl.js](https://github.com/godaddy/gasket/commit/d43cdd46db2f0cfbcf4c2f472a504fcb1033a4fe)
- [Add json assertion to dynamic import](https://github.com/godaddy/gasket/commit/4785d480fac4170581e3a03fa5dc3e6daeb01ee0)
- [Add ESM flag to jest for dynamic import](https://github.com/godaddy/gasket/commit/4b96a0a16ece041353f558be22d09c930f32388f)
- [Add config for experimental import attributes](https://github.com/godaddy/gasket/pull/975/commits/02419526b3b827e374feac9d5b67752515a26932)

## Changelog
- Tune getIntlManager to import intl.js
- Add json assertion to dynamic import
- Add ESM flag to jest for dynamic import
- Add config for experimental import attributes

## Test Plan
Updated

### CODE REVIEW

File: packages/gasket-plugin-intl/lib/index.d.ts [hunk 1]
```diff
@@ -55,13 +56,16 @@ declare module '@gasket/data' {
...
+      /* dynamic import attributes toggle */
+      experimentalImportAttributes?: boolean;
     };
   }
 }
 
 declare module 'create-gasket-app' {
   export interface CreateContext {
     hasGasketIntl?: boolean;
+    experimentalImportAttributes?: boolean;
```
[the remark this answers is not in the document]
Comment (mmason2-godaddy) [reply packages/gasket-plugin-intl/lib/index.d.ts:68]: I was opting for the `experimentalImportAttributes = true;` in the preset but I can offload that to the plugin directly

File: packages/gasket-plugin-intl/lib/index.d.ts [hunk 2]
```diff
@@ -55,6 +56,8 @@ declare module '@gasket/data' {
   export interface GasketData {
     intl?: {
       locale: string;
+      /* dynamic import attributes toggle */
+      experimentalImportAttributes?: boolean;
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-intl/lib/index.d.ts:60]: You're not adding this via the publicGasketData hook, and I don't see it as necessary anyhoo
```suggestion
```

**Review by agerard-godaddy — APPROVED** [review]
Small tweak then 👍

**Review by jpina1-godaddy — APPROVED** [review]

File: packages/gasket-plugin-intl/lib/build-manifest.js [hunk 3]
```diff
@@ -50,6 +50,10 @@ module.exports = async function buildManifest(gasket, options = {}) {
       const keyName = importName
         .replace(/\.json$/, '')
         .replace('./', '');
+      if (gasket.config.intl.experimentalImportAttributes) {
```
Comment (aryasaatvik) [inline packages/gasket-plugin-intl/lib/build-manifest.js:53]: y'all should create a top level experimental/future flag which uses import asserts. 

import as/from are executed at build time. createRequire is  executed at runtime which makes it harder to run gasket in serverless environments

**Pull request merged**